### PR TITLE
Chris/View Binding

### PR DESCRIPTION
**Pull request by ckdesir**
## Overview

Migrates to View Binding

## Changes Made

- [x] Migrate to View Binding (phase out kotlin-android-extensions)
- [x] Make layout names and ids more intuitive
- [x] Removes unused files
- [x] Combine both HomeFollowingAdapter and HomeOtherArticlesAdapter -> HomeArticlesAdapter
- [x] Gradle updates 

## Next Steps 

- [x] Firebase Analytics
- [x] Onboarding Update 
- [x] No internet screen (#22)
- [x] String.xml flesh-out

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,11 +1,11 @@
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
+apply plugin: 'kotlin-parcelize'
```
Comment (kevinsun-dev) [inline app/build.gradle:3]: Ayy, I approve.

File: app/src/main/java/com/cornellappdev/volume/PublicationProfileActivity.kt [hunk 2]
```diff
@@ -143,40 +114,40 @@ class PublicationProfileActivity : AppCompatActivity() {
...
+                    binding.clFbHolder.visibility = View.GONE
                 }
-                profile_name.text = publication.name
-                profile_shoutouts.text = publication.shoutouts.toInt().toString() + " shoutouts"
-                profile_desc.text = publication.bio
-                Picasso.get().load(publication.backgroundImageURL).into(profile_banner)
-                Picasso.get().load(publication.profileImageURL).into(profile_logo)
+                binding.tvName.text = publication.name
+                binding.tvShoutoutCount.text =
+                        publication.shoutouts.toInt().toString() + " shoutouts"
```
Comment (kevinsun-dev) [inline app/src/main/java/com/cornellappdev/volume/PublicationProfileActivity.kt:139]: In a future PR, replace these with format strings from the .xml file.

File: app/src/main/java/com/cornellappdev/volume/adapters/HomeArticlesAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        val currentItem = articles[position]
+        holder.binding.tvArticleTitle.text = currentItem.title
+        Article.applyNSFWFilter(currentItem, holder.binding.tvArticleTitle)
+        if (!currentItem.imageURL.isNullOrBlank()) {
+            holder.binding.ivArticleImage.visibility = View.VISIBLE
+            Picasso.get().load(currentItem.imageURL).into(holder.binding.ivArticleImage)
+        }
+        Article.setCorrectDateText(currentItem, holder.binding.tvTimePosted)
+        holder.binding.tvShoutoutCount.text =
+                currentItem.shoutouts?.toInt().toString() + " shout-outs"
```
Comment (kevinsun-dev) [inline app/src/main/java/com/cornellappdev/volume/adapters/HomeArticlesAdapter.kt:37]: Shout-outs here compared to " shoutouts" before. Is this intentional?

File: app/src/main/java/com/cornellappdev/volume/adapters/MorePublicationsAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        holder.binding.btnFollow.setOnClickListener {
+            if (holder.binding.btnFollow.drawable.constantState == ContextCompat.getDrawable(it.context,
+                            R.drawable.ic_followplussvg)!!.constantState) {
+                holder.binding.btnFollow.setImageResource(R.drawable.ic_followchecksvg)
+                val tempSet =
+                        prefUtils.getStringSet("following", mutableSetOf())?.toMutableSet()
+                if (tempSet != null) {
+                    tempSet.add(currentItem.id)
+                    prefUtils.save("following", tempSet)
```
Comment (kevinsun-dev) [inline app/src/main/java/com/cornellappdev/volume/adapters/MorePublicationsAdapter.kt:62]: Used "following" as a string in one file, should probably go make it a constant somewhere.

File: app/src/main/java/com/cornellappdev/volume/adapters/MorePublicationsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                if (tempSet != null) {
+                    tempSet.remove(currentItem.id)
+                    prefUtils.save("following", tempSet)
+                }
+            }
+        }
+
+        holder.binding.clPublicationLayout.setOnClickListener { view ->
+            val intent = Intent(view.context, PublicationProfileActivity::class.java)
+            intent.putExtra("publication", currentItem)
```
Comment (kevinsun-dev) [inline app/src/main/java/com/cornellappdev/volume/adapters/MorePublicationsAdapter.kt:77]: Same with publication. In general, consider making these keys constant.

File: app/src/main/java/com/cornellappdev/volume/adapters/SavedArticlesAdapter.kt [hunk 6]
```diff
@@ -5,59 +5,41 @@ import android.os.Build
...
         }
-        Article.setCorrectDateText(currentItem, holder.postTime)
-        holder.shoutoutCount.text = currentItem.shoutouts?.toInt().toString() + " shout-outs"
-        holder.pubName.text = currentItem.publication?.name
-
-        holder.layoutMain.setOnClickListener {
-            val intent = Intent(holder.layout.context, MainActivity::class.java)
+        Article.setCorrectDateText(currentItem, holder.binding.tvTimePosted)
+        holder.binding.tvShoutoutCount.text =
+                currentItem.shoutouts?.toInt().toString() + " shout-outs"
```
Comment (kevinsun-dev) [inline app/src/main/java/com/cornellappdev/volume/adapters/SavedArticlesAdapter.kt:37]: Same question. "shout-outs" or "shoutouts"?

**Review by kevinsun-dev — APPROVED** [review]
Good migration. ViewBinding is absolutely the future of Android development and we should move in that direction.

File: app/src/main/java/com/cornellappdev/volume/adapters/HomeArticlesAdapter.kt (hunk 3 above)
Comment (ckdesir) [reply app/src/main/java/com/cornellappdev/volume/adapters/HomeArticlesAdapter.kt:37]: had to double check designs, i think shout-outs is intentional

**Comment by ckdesir** [comment]
ViewBinding def way better!

File: app/src/main/java/com/cornellappdev/volume/PublicationProfileActivity.kt (hunk 2 above)
Comment (ckdesir) [reply app/src/main/java/com/cornellappdev/volume/PublicationProfileActivity.kt:139]: yup soon to come

**Pull request merged**